File: submission/12017800/code/08_merge_coop/VincentFunction.java
public class VincentFunction implements OptimizationFunction {
	public double getFitness(Individual individual) {
		double sumPart = 0;
		double iValue;

		for (int i = 0; i < individual.getNumCells(); i++) {
			iValue = individual.getCell(i).getValue();
			sumPart += Math.sin(10 * Math.sqrt(iValue));
		}

		return -1.0 * (1 + sumPart);
	}

	public double getMin() {
		return 0.25;
	}
	
	public double getMax() {
		return 10.0;
	}

	@Override
	public String toString() {
		return "Vincent [" + getMin() + "," + getMax() + "]";
	}
}

File: submission/12017800/code/02_basic_coop/vincentFunction.cpp
#include "vincentFunction.h"

#include <cfloat>
#include <cmath>

using namespace std;

VincentFunction::VincentFunction() {

}

VincentFunction::~VincentFunction() {

}

long double VincentFunction::getFitness(Individual* individual) {
	long double sum = 0;
	long double iValue = 0;

	for (int i = 0; i < individual->getNumCells(); i++) {
		iValue = (*individual)[i].getValue();
		sum += sin(10.0 * sqrt(iValue));
	}

	return -1.0 * (1.0 + sum);
}

float VincentFunction::getMin() {
	return 0.25;
}

float VincentFunction::getMax() {
	return 10;
}

string VincentFunction::getName() const {
	return "Vincent";
}

ostream& operator<<(ostream& os, const VincentFunction& function) {
	os << function.getName();
	return os;
}